lib/absinthe/schema.ex

```diff
@@ -575,6 +576,21 @@ defmodule Absinthe.Schema do
     # successfully compile the schema.
     {:ok, bp, _} = Absinthe.Pipeline.run(schema.__absinthe_blueprint__, pipeline)
 
+    bp =
+      if Keyword.get(opts, :include_disconnected, false) do
+        bp
+      else
+        Map.update!(bp, :schema_definitions, fn schema_defs ->
+          for schema_def <- schema_defs do
+            Map.update!(schema_def, :type_definitions, fn type_defs ->
+              Enum.filter(type_defs, fn type_def ->
+                type_def.__private__[:__absinthe_used__]
```

Comment:
The concept of "referenced" is in the spec, so maybe we could elevate it into the struct itself vs sticking it in `__private__`?

Reply:
@binaryseed my concern with making it a first class field is that it isn't really something you should mess with yourself. Whether something is referenced or not requires a holistic look at the schema using relatively complex rules.

If people start needing it regularly then we can look at exposing it more directly. Right now it's a pretty low level detail.

Reply:
makes sense